packages/terra-pills/src/private/_RollupPill.jsx

```diff
@@ -0,0 +1,95 @@
+import React, {
+  useRef,
+} from 'react';
+import {
+  KEY_SPACE, KEY_RETURN,
+} from 'keycode-js';
+import { injectIntl } from 'react-intl';
+import classNamesBind from 'classnames/bind';
+import PropTypes from 'prop-types';
+import ThemeContext from 'terra-theme-context';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
+import styles from './Pill.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * Determines if the Pill List is rolled up or not.
+   */
+  isCollapsed: PropTypes.bool,
+  /**
+   * Callback function triggered on click/key press of the roll-up pill
+   */
+  onSelectRollUp: PropTypes.func,
+  /**
+   * Number of pills that are rolled up.
+   */
+  rollupCount: PropTypes.number,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+const RollUpPill = (props) => {
+  const {
+    isCollapsed, onSelectRollUp, intl, rollupCount,
+  } = props;
+  const rollUpPillRef = useRef();
+  const theme = React.useContext(ThemeContext);
+
+  const handleOnSelectRollUp = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+    onSelectRollUp(event);
+  };
+
+  const handleRollUpPillKeyDown = (event) => {
+    rollUpPillRef.current.setAttribute('data-terra-rollup-pill-show-focus-styles', 'true');
+    if (event.keyCode === KEY_RETURN || event.keyCode === KEY_SPACE) {
+      event.preventDefault();
+      event.stopPropagation();
+      onSelectRollUp(event);
+    }
+  };
+
+  const handleRollUpPillMouseDown = () => {
+    rollUpPillRef.current.setAttribute('data-terra-rollup-pill-show-focus-styles', 'false');
+  };
+
+  const handleOnBlur = () => {
+    rollUpPillRef.current.setAttribute('data-terra-rollup-pill-show-focus-styles', 'true');
+  };
+
+  const rollUpLabel = isCollapsed && rollupCount > 0 ? intl.formatMessage({ id: 'Terra.pills.label.rollupPill' }, { pillsNotVisibleCount: rollupCount }) : intl.formatMessage({ id: 'Terra.pills.label.showLess' });
+  const rollUpHint = isCollapsed ? intl.formatMessage({ id: 'Terra.pills.hint.rollupPill' }, { pillsNotVisibleCount: rollupCount }) : intl.formatMessage({ id: 'Terra.pills.hint.showLess' });
+  return (
+    <div
+      role="listitem"
+      className={cx('pill-list-item')}
+    >
+      <button
+        id="rollup-pill"
```

Comment:
This id should be unique.

Reply:
I may be missing it, but where is an id for this needed? Could a class or data attribute be used instead?

Reply:
Changed cfb6aad5b38a9957f1449aa2f75bbf5d001c3c96

The [Id](https://github.com/cerner/terra-framework/blob/terra-filter-pills/packages/terra-pills/src/FilterPills.jsx#L230) is fetched here which is later used to get the node for focusing.